presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveType.java

```diff
@@ -115,7 +123,12 @@ public TypeInfo getTypeInfo()
 
     public TypeSignature getTypeSignature()
     {
-        return getTypeSignature(typeInfo);
+        return semanticTypeSignature.orElse(getTypeSignature(typeInfo));
+    }
+
+    public HiveType copyWithoutSemanticType()
```

Comment:
suggestion: I would inline this method, as it doesn't make a lot of sense as a member of this class.